tar/bsdtar.c

```diff
@@ -74,20 +74,19 @@ __FBSDID("$FreeBSD: src/usr.bin/tar/bsdtar.c,v 1.93 2008/11/08 04:43:24 kientzle
  * Per POSIX.1-1988, tar defaults to reading/writing archives to/from
  * the default tape device for the system.  Pick something reasonable here.
  */
-#ifdef __linux
-#define	_PATH_DEFTAPE "/dev/st0"
-#endif
-#if defined(_WIN32) && !defined(__CYGWIN__)
-#define	_PATH_DEFTAPE "\\\\.\\tape0"
-#endif
-#if defined(__APPLE__)
+#if ARCHIVE_VERSION_NUMBER >= 4000000
 #undef _PATH_DEFTAPE
-#define	_PATH_DEFTAPE "-"  /* Mac OS has no tape support, default to stdio. */
+#elif defined(__linux)
+#define _PATH_DEFTAPE "/dev/st0"
+#elif defined(_WIN32) && !defined(__CYGWIN__)
+#define _PATH_DEFTAPE "\\\\.\\tape0"
+#elif defined(__APPLE__)
+#undef _PATH_DEFTAPE  /* Don't use a default tape device on macOS */
+#elif defined(__FreeBSD__)
+#define _PATH_DEFTAPE "/dev/tape"
```

Comment:
@kientzle it looks like some FreeBSD configurations already defines `_PATH_DEFTAPE`, so I guess we should drop this, do you agree?

Reply:
As a safer approach I think would be better
```
#elif defined(__FreeBSD__) && !defined(_PATH_DEFTAPE)
```

Reply:
Hmmm.... Yeah, we should probably honor it if it's already defined, and not just on FreeBSD.  How about this instead?
```
#if ARCHIVE_VERSION_NUMBER < 4000000 && !defined(_PATH_DEFTAPE)
// Libarchive 4.0 and later will NOT define _PATH_DEFTAPE
// but will honor it if it's set in the build.
// Until then, we'll continue to set it by default on certain platforms:
#if defined(__linux)
#define _PATH_DEFTAPE "/dev/st0"
#elif defined(_WIN32) && !defined(__CYGWIN__)
#define _PATH_DEFTAPE "\\\\.\\tape0"
#elif defined(__FreeBSD__)
#define _PATH_DEFTAPE "/dev/tape"
#endif
#endif 
```

Reply:
Thank you for your prompt reply. I like your proposal, let me apply it.